mvlearn/decomposition/groupica.py

```diff
@@ -0,0 +1,326 @@
+            )
+            sources = sources.T
+        if K is not None:
+            self.components_ = np.dot(W, K)
+        else:
+            self.components_ = W
+        self.mixing_ = linalg.pinv(self.components_)
+        # Compute individual unmixing matrices by least-squares
+        self.individual_mixing_ = []
+        self.individual_components_ = []
+        sources_pinv = linalg.pinv(sources)
+        for X, mean in zip(Xs, self.means_):
+            lstq_solution = np.dot(sources_pinv, X - mean)
+            self.individual_components_.append(linalg.pinv(lstq_solution).T)
+            self.individual_mixing_.append(lstq_solution.T)
+        self.n_components_ = gpca.n_components_
+        self.n_features_ = gpca.n_features_
+        self.n_samples_ = gpca.n_samples_
+        self.n_subjects_ = gpca.n_subjects_
+        return sources
+
+    def fit_transform(self, Xs, y=None):
+        """
+        Fit to the data and transform the data into sources
+
+        Parameters
+        ----------
+        Xs : list of array-likes or numpy.ndarray
+             - Xs length: n_views
+             - Xs[i] shape: (n_samples, n_features_i)
+        y : array, shape (n_samples,), optional
+
+        Returns
+        -------
+        X_transformed : list of array-likes or numpy.ndarray
+            The transformed data.
+            If `multiple_outputs` is True, it is a list with the estimated
+            individual sources.
+            If `multiple_outputs` is False, it is a single array containing the
+            shared sources.
+        """
+        sources = self._fit(Xs, y)
+        if self.multiple_outputs:
+            return self.transform(Xs)
+        else:
+            return sources
+
+    def fit(self, Xs, y=None):
+        r"""Fit the model with Xs.
+        Parameters
+        ----------
+        Xs: list of array-likes
+            - Xs shape: (n_views,)
+            - Xs[i] shape: (n_samples, n_features_i)
+        y : None
+            Ignored variable.
+        Returns
+        -------
+        self : object
+            Returns the instance itself.
+        """
+        self._fit(Xs, y)
+        return self
+
+    def transform(self, Xs, y=None):
+        r"""
+        A method to fit model to multiview data.
+
+        Parameters
+        ----------
+        Xs: list of array-likes
+            - Xs shape: (n_views,)
+            - Xs[i] shape: (n_samples, n_features_i)
+        y : array, shape (n_samples,), optional
+
+        Returns
+        -------
+        X_transformed : list of array-likes or numpy.ndarray
+            The transformed data.
+            If `multiple_outputs` is True, it is a list with the estimated
+            individual sources.
+            If `multiple_outputs` is False, it is a single array containing the
+            shared sources.
+        """
+        check_is_fitted(self)
+        Xs = check_Xs(Xs, copy=True)
+
+        if self.multiple_outputs:
+            return [
```

Comment:
should the output of a `.transform` be a list of 2 d arrays or a 3d array?

Reply:
We typically do a list of 2d arrays. 1) not guaranteed the shapes across views will allow for a 3d array. 2) the interpretation of different views is more clear.

Reply:
Ok, so this code is fine.